crates/db_views/src/post_view.rs

```diff
@@ -305,16 +305,18 @@ impl<'a> PostQuery<'a> {
       ))
       .into_boxed();
 
-    // Hide deleted and removed for non-admins or mods
-    // TODO This eventually needs to show posts where you are the creator
-    if !self.is_mod_or_admin.unwrap_or(false) {
+    if !self.show_deleted.unwrap_or(false) {
       query = query
-        .filter(community::removed.eq(false))
         .filter(community::deleted.eq(false))
-        .filter(post::removed.eq(false))
         .filter(post::deleted.eq(false));
     }
 
+    if !self.show_removed.unwrap_or(false) {
+      query = query
+        .filter(community::removed.eq(false))
+        .filter(post::removed.eq(false));
+    }
```

Comment:
Its a bit unclear what to do if the community was removed/deleted, we can revisit this if issues are reported in the future.

Reply:
The difficulty with this one, and why this really needs a SQL solution (that's better than my old one), is that the posts fetched are a collection of posts that *might be* yours + others.

With admins and mods, we can just show all deleted and removed. When its the creator, we need to do a join + filter.

Reply:
With the current usage this is no problem as show_removed and show_deleted are only set on the user profile, so it will only include posts authored by that user. But if it ever gets set from somewhere else that will cause problems. As an admin I also dont want to see deleted and removed posts all over the frontpage, its pretty annoying.

As a kind of solution/workaround we could add a check that show_removed/show_deleted can only be true if creator_id is Some?

Reply:
No because `!self.show_removed.unwrap_or(false)` with `show_removed = None` evaluates to `!false` which is true so the code in if is executed. Admittedly its confusing but its also covered by tests.

I changed the code now to avoid explicit show_removed/show_deleted parameters. If this looks better to you, I will also fix the tests.

Reply:
Yes it does look better.